lib/features/email/data/network/email_api.dart

```diff
@@ -687,4 +694,52 @@ class EmailAPI with HandleSetErrorMixin {
       throw NotFoundEmailException();
     }
   }
+
+  Future<EmailRecoveryAction> restoreDeletedMessage(RestoredDeletedMessageRequest restoredDeletedMessageRequest) async {
+    final processingInvocation = ProcessingInvocation();
+    final requestBuilder = JmapRequestBuilder(_httpClient, processingInvocation);
+
+    final emailRecoveryActionSetMethod = SetEmailRecoveryActionMethod()
+      ..addCreate(
+        restoredDeletedMessageRequest.createRequestId,
+        restoredDeletedMessageRequest.emailRecoveryAction
+      );
+    final emailRecoveryActionSetInvocation = requestBuilder.invocation(emailRecoveryActionSetMethod);
+    final response = await (requestBuilder
+        ..usings(emailRecoveryActionSetMethod.requiredCapabilities))
+      .build()
+      .execute();
+    
+    final emailRecoveryActionSetResponse = response.parse<SetEmailRecoveryActionResponse>(
+      emailRecoveryActionSetInvocation.methodCallId,
+      SetEmailRecoveryActionResponse.deserialize
+    );
+
+    return emailRecoveryActionSetResponse!.created![restoredDeletedMessageRequest.createRequestId]!;
+  }
+
+  Future<EmailRecoveryAction> getRestoredDeletedMessage(EmailRecoveryActionId emailRecoveryActionId) {
+    final processingInvocation = ProcessingInvocation();
+    final requestBuilder = JmapRequestBuilder(_httpClient, processingInvocation);
+
+    final getEmailRecoveryActionMethod = GetEmailRecoveryActionMethod()
+      ..addIds({emailRecoveryActionId.id});
+    final getEmailRecoveryActionInvocation = requestBuilder.invocation(getEmailRecoveryActionMethod);
+    return (requestBuilder
+        ..usings(getEmailRecoveryActionMethod.requiredCapabilities))
+      .build()
+      .execute()
+      .then((response) {
+        final resultList = response.parse<GetEmailRecoveryActionResponse>(
+          getEmailRecoveryActionInvocation.methodCallId,
+          GetEmailRecoveryActionResponse.deserialize
+        );
+
+        if (resultList?.list.isNotEmpty == true) {
+          return resultList!.list.firstWhere((element) => element.id == emailRecoveryActionId);
+        } else {
+          throw NotFoundEmailRecoveryActionException();
+        }
+      });
```

Comment:
We should unify the way we write code in handling api from jmap. Please make like the following image

<img width="798" alt="Screenshot 2024-01-02 at 10 44 22" src="https://github.com/linagora/tmail-flutter/assets/80730648/11a81cff-c5fb-489d-b283-57583275e02d">